pkg/controller/cassandradatacenter/cassandradatacenter_controller.go

```diff
@@ -151,14 +167,112 @@ func (r *ReconcileCassandraDataCenter) Reconcile(request reconcile.Request) (rec
 		return reconcile.Result{}, err
 	}
 
-	// TODO:
 	_, _, _ = nodesService, seedNodesService, statefulSet
 
 	rctx.logger.Info("CassandraDataCenter reconciliation complete.")
 
 	return reconcile.Result{}, nil
 }
 
+func getOperatorDefaultConfig(c client.Client, namespace string, defaultOperatorConfigMapName string) (corev1.ConfigMap, error) {
+	defaultConfig := &corev1.ConfigMap{}
+
+	if err := c.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: defaultOperatorConfigMapName}, defaultConfig); err != nil {
+		if k8sErrors.IsNotFound(err) {
+			// if it is not found, just return empty config map
+			return *defaultConfig, nil
+		} else {
+			// otherwise there is something shady going on
+			return *defaultConfig, err
+		}
+	} else {
+		return *defaultConfig, nil
+	}
+}
+
+func populateUnsetFields(instance *cassandraoperatorv1alpha1.CassandraDataCenter, configMap corev1.ConfigMap) (bool, error) {
+	populated := false
```

Comment:
I would drop `populated` as if we ever not populate a field, we return false explicitly.